app/subscriber/src/features/search-page/SearchPage.tsx

```diff
@@ -104,11 +112,14 @@ export const SearchPage: React.FC<ISearchType> = ({ showAdvanced }) => {
     fetchResults(query, filter.searchUnpublished);
   }, [fetchResults, filter, genQuery]);
 
-  const executeSearch = React.useCallback(async () => {
-    const settings = filterFormat(filter);
-    const query = genQuery(settings);
-    fetchResults(query, filter.searchUnpublished);
-  }, [fetchResults, filter, genQuery]);
+  const executeSearch = React.useCallback(
+    async (filter: IFilterSettingsModel) => {
```

Comment:
Understanding state and dependencies is important for making functional event handlers.